File: javascript/wordy/wordy.js
const find_operator = (str) => {
  switch (str) {
    case 'plus':
      return '+' ;
    case 'minus':
      return '-' ;
    case 'multiplied':
      return '*' ;
    case 'divided':
      return '/' ;
  }
};

export class WordProblem {
  constructor(question) {
    this.operation = this.clean(question);
  }

  clean(str) {
    return str.replace(/What is|by|\?/g, '').split(' ').map(el => {
     return find_operator(el) ? find_operator(el) : el;
    }).join(' ');
  }

  answer() {
    return eval(this.operation);
  }
}

File: ruby/wordy/wordy.rb
class WordProblem
  def initialize(question)
    @operation = question.gsub('What is ', '')[0..-2]
  end

  def answer
    operators = {
      'plus' => '+',
      'minus' => '-',
      'multiplied by' => '*',
      'divided by' => '/'
    }

    replaced = @operation.gsub(/\b(plus|minus|multiplied by|divided by)\b/, operators)
    eval(replaced)

  end
end
